src/App.php

```diff
@@ -265,13 +265,23 @@ protected function configureRoutes( Slim $slim ) {
 				$middleware['inject-user'], $middleware['set-environment'],
 			function () use ( $slim, $middleware, $routeConditions ) {
 				$slim->get( '', function() use ( $slim ) {
-					// Redirect root-URL requests to English.
-					$currentLang = $slim->i18nContext->getCurrentLanguage();
-					$slim->redirectTo( 'home', [ 'wikiLang' => $currentLang ] );
+					// See if we have a cookie indicating last version used
+					if ( isset( $_COOKIE['copypatrolLang'] ) ) {
+						$slim->redirectTo( 'home', [ 'wikiLang' => $_COOKIE['copypatrolLang'] ] );
+					} else {
+						// If no cookie, check if we support i18nContext's default language
+						$lang = $slim->i18nContext->getCurrentLanguage();
+						if ( in_array( $lang, [ 'en', 'fr' ] ) ) {
+							$slim->redirectTo( 'home', [ 'wikiLang' => $lang ] );
+						}
+						// We don't support i18nContext's current language, so redirect to en version
+						$slim->redirectTo( 'home', [ 'wikiLang' => 'en' ] );
+					}
```

Comment:
It does look less understandable. I'd prefer sticking with the current version. :)